### PR TITLE
feature/foreign keys - issue #966

### PR DESCRIPTION
**Pull request by jocull**
This is a big project and I am not sure I did this right. I added a way to add foreign keys to match how they were being dropped. Postgres dialect needs to be added, but I did MySQL and MSSQL (SQLite doesn't seem to apply / it's broken beyond me fixing it)

### CODE REVIEW

**Comment by jocull** [comment]
I'm not sure how to reference issue #966 from this, so this is me trying again...

**Comment by mickhansen** [comment]
There's some lintin errors.

File: lib/dialects/mssql/query-generator.js [hunk 1]
```diff
@@ -508,6 +508,15 @@ var QueryGenerator = {
...
   },
+  
+  addForeignKeyQuery: function(targetTableName, targetColumnName, foreignKeyName,
+    referenceTableName, referenceColumnName) {
+      
+    return "ALTER TABLE " + this.quoteTable(targetTableName) + " WITH CHECK " +
+      "ADD CONSTRAINT " + this.quoteIdentifier(foreignKeyName) + " " +
+      "FOREIGN KEY (" + this.quoteIdentifier(targetColumnName) + ") " +
+      "REFERENCES " + this.quoteTable(targetTableName) + "(" + this.quoteIdentifier(targetColumnName) + ");"
+  },
```
Comment (mickhansen) [inline lib/dialects/mssql/query-generator.js]: Would support for `ON UPDATE` and `ON DELETE` not also be needed?

**Comment by mickhansen** [comment]
@sladebot This PR does not currently address PG

**Comment by jocull** [comment]
I need to investigate again and see how `ON UPDATE` and `ON DELETE` are
handled elsewhere for consistency. Anyone's welcome to join in and
contribute on my pull request - I just wanted to start the movement on
getting this resolved finally. If someone else could handle PG especially
that would be good.

**Pull request closed without merging**